Data Structures/Linked Lists/C++/linkedlist.cpp

```diff
@@ -0,0 +1,135 @@
+#include<iostream>
+using namespace std;
+
+int l;
+void FindLocation();
+struct student{
+  int data;
+  student* link; 
+} *start,*loc,*ptr1;
+
+void insertion_begin(int y){
+  student* temp=new student();
+ temp->data=y;
+ temp->link=start;
+ start=temp;
+}
+void insertion_end(int y){
+ student* temp=new student();
+ temp->data=y;
+ if(start==NULL){
+ temp->link=NULL;
+ start=temp;
+}
+ else{
+ student* ptr=start;
+ while(ptr->link!=NULL) 
+ ptr=ptr->link;
+ temp->link=NULL;
+ ptr->link=temp;
+}
+}
+void insertion_after(int y){
+	
+	student* temp=new student();
+	temp->data=y;
+	cout<<endl<<"enter the element after which to be inserted: ";                   cin>>l;
+	FindLocation();
+	
+   		if(loc==NULL){
+		  temp->link=start;
+		  start=temp;
+		}
+     	  	else{
+		  temp->link=loc->link;
+		  loc->link=temp;
+		}
+}
+void insertion_at(){
+	int a,count=1,item;
+	cout<<"enter the location at which data to be inserted: ";
+	cin>>a;
+	cout<<endl<<"enter the element to be inserted: ";
+	cin>>item;
+	student* temp=new student();
+	student* ptr=start;
+	while(ptr->link!=NULL){
+	if(count==a){
+	  temp->data=item;
+	  temp->link=ptr->link;
+	  ptr->link=temp;
+   	}
+	 else{
+		count++;
+		ptr=ptr->link;
+
+	  }
+}
+
+}
+void FindLocation(){
+	 ptr1=start;
+	while(ptr1!=NULL){
+	if(ptr1->data==l){
+	  loc=ptr1;
+	  return;
+	}
+	else
+	 ptr1=ptr1->link;
+	}
+   loc=NULL;
+   
+}
+void display()
+{
+  student* temp=start;
+  while(temp!=NULL){
+  cout<<temp->data<<" ";
+  temp=temp->link;
+}
+cout<<endl<<endl;
+}
+int main()
+{
+	start=NULL;
+	int i,n,x,k,m;
+	 cout<<endl<<"Enter the number of elements: ";
```

Comment:
We shouldn't take the "number of elements" input here, isn't it? Can you shift it inside the `case 1` and `case 2` blocks if the `n` variable is being used inside that cases only?